code/game/machinery/_machinery.dm

```diff
@@ -336,44 +336,52 @@
-		if(!silicon || !(interaction_flags_machine & INTERACT_MACHINE_OPEN_SILICON))
-			return FALSE
+	if(isAdminGhostAI(user))
+		return TRUE //if you're an admin, you probably know what you're doing (or at least have permission to do what you're doing)
 
-	if(silicon || isAdminGhostAI(user)) // If we are an AI or adminghsot, make sure the machine allows silicons to interact
-		if(!(interaction_flags_machine & INTERACT_MACHINE_ALLOW_SILICON))
-			return FALSE
+	if(!isliving(user))
+		return FALSE //no ghosts in the machine allowed, sorry
+
+	var/mob/living/L = user
 
 	var/is_dextrous = FALSE
 	if(isanimal(user))
 		var/mob/living/simple_animal/user_as_animal = user
 		if (user_as_animal.dextrous)
 			is_dextrous = TRUE
 
-	if(is_dextrous || user.can_hold_items()) // If we are a living mob with hand slots or a dextrous simple animal.
-		var/mob/living/L = user
+	if(!(issilicon(user) || is_dextrous || user.can_hold_items()))
+		return FALSE //spiders gtfo
+	
+	if(issilicon(user)) // If we are a silicon, make sure the machine allows silicons to interact with it
+		if(!(interaction_flags_machine & INTERACT_MACHINE_ALLOW_SILICON))
+			return FALSE
 
-		if(interaction_flags_machine & INTERACT_MACHINE_REQUIRES_SILICON) // First make sure the machine doesn't require silicon interaction
+		if(panel_open && !(interaction_flags_machine & INTERACT_MACHINE_OPEN) && !(interaction_flags_machine & INTERACT_MACHINE_OPEN_SILICON))
 			return FALSE
 
-		if(interaction_flags_machine & INTERACT_MACHINE_REQUIRES_SIGHT)
-			if(user.is_blind())
-				to_chat(user, "<span class='warning'>This machine requires sight to use.</span>")
-				return FALSE
+		return TRUE //silicons don't care about petty mortal concerns like needing to be next to a machine to use it
 
-		if(!Adjacent(user)) // Next make sure we are next to the machine unless we have telekinesis
-			var/mob/living/carbon/H = L
-			if(!(istype(H) && H.has_dna() && H.dna.check_mutation(TK)))
-				return FALSE
+	if(L.incapacitated()) //idk why silicons aren't supposed to care about incapacitation when interacting with machines, but it was apparently like this before
+		return FALSE
 
-		if(L.incapacitated()) // Finally make sure we aren't incapacitated
+	if(interaction_flags_machine & INTERACT_MACHINE_REQUIRES_SIGHT)
+		if(user.is_blind())
+			to_chat(user, "<span class='warning'>This machine requires sight to use.</span>")
 			return FALSE
 
-	else // If we aren't a silicon, living, or admin ghost, bad!
+	if(panel_open && !(interaction_flags_machine & INTERACT_MACHINE_OPEN))
+		return FALSE
+
+	if(interaction_flags_machine & INTERACT_MACHINE_REQUIRES_SILICON) //if the user was a silicon, we'd have returned out earlier, so the user must not be a silicon
 		return FALSE
 
-	return TRUE // If we pass all these checks, woohoo! We can interact
+	if(!Adjacent(user)) // Next make sure we are next to the machine unless we have telekinesis
+		var/mob/living/carbon/H = L
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

Reply:
FINE